docs/resources/sso_settings.md

```diff
@@ -72,17 +72,87 @@ resource "grafana_sso_settings" "saml_sso_settings" {
 
 ### Required
 
-- `provider_name` (String) The name of the SSO provider. Supported values: github, gitlab, google, azuread, okta, generic_oauth, saml.
+- `provider_name` (String) The name of the SSO provider. Supported values: github, gitlab, google, azuread, okta, generic_oauth, saml, ldap.
 
 ### Optional
 
+- `ldap_settings` (Block Set, Max: 1) The LDAP settings set. Required for the ldap provider. (see [below for nested schema](#nestedblock--ldap_settings))
 - `oauth2_settings` (Block Set, Max: 1) The OAuth2 settings set. Required for github, gitlab, google, azuread, okta, generic_oauth providers. (see [below for nested schema](#nestedblock--oauth2_settings))
 - `saml_settings` (Block Set, Max: 1) The SAML settings set. Required for the saml provider. (see [below for nested schema](#nestedblock--saml_settings))
 
 ### Read-Only
 
 - `id` (String) The ID of this resource.
 
+<a id="nestedblock--ldap_settings"></a>
+### Nested Schema for `ldap_settings`
+
+Required:
+
+- `config` (Block List, Min: 1, Max: 1) The LDAP configuration. (see [below for nested schema](#nestedblock--ldap_settings--config))
+
+Optional:
+
+- `allow_sign_up` (Boolean) Whether to allow new Grafana user creation through LDAP login. If set to false, then only existing Grafana users can log in with LDAP.
+- `enabled` (Boolean) Define whether this configuration is enabled for LDAP. Defaults to `true`.
+- `skip_org_role_sync` (Boolean) Prevent synchronizing users’ organization roles from LDAP.
+
+<a id="nestedblock--ldap_settings--config"></a>
+### Nested Schema for `ldap_settings.config`
+
+Required:
+
+- `servers` (Block List, Min: 1) The LDAP servers configuration. (see [below for nested schema](#nestedblock--ldap_settings--config--servers))
+
+<a id="nestedblock--ldap_settings--config--servers"></a>
+### Nested Schema for `ldap_settings.config.servers`
+
+Required:
+
+- `host` (String) The LDAP server host.
+- `search_base_dns` (List of String) An array of base DNs to search through.
+- `search_filter` (String) The user search filter, for example "(cn=%s)" or "(sAMAccountName=%s)" or "(uid=%s)".
+
+Optional:
+
+- `attributes` (Map of String) The LDAP server attributes. The following attributes can be configured: email, member_of, name, surname, username.
```

Comment:
I'm not 100% sure, but I think either the `email` or `username` must be specified.

Reply:
This is not enforced on the server side and I think we shouldn't do it on the client.